src/Fuzz.elm

```diff
@@ -357,13 +365,35 @@ listShrinkHelp listOfTrees =
         root =
             List.map RoseTree.root listOfTrees
 
+        dropFirstHalf : List (RoseTree a) -> RoseTree (List a)
+        dropFirstHalf list_ =
+            List.drop (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        dropSecondHalf : List (RoseTree a) -> RoseTree (List a)
+        dropSecondHalf list_ =
+            List.take (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        halved : LazyList (RoseTree (List a))
+        halved =
+            if n >= 4 then
+                Lazy.lazy <|
+                    \_ ->
+                        Lazy.List.fromList [ dropFirstHalf listOfTrees, dropSecondHalf listOfTrees ]
+                            |> Lazy.force
+            else
+                -- For lists of three elements or less, halving and removing a single element is often going to be the same thing.
+                -- To prevent us from attempting the same shrunken value twice, we're disabling the halving strategy for these small lists.
```

Comment:
Could we move this comment up to the first line of the if-statement, so that you have it right there when you wonder what that magic number is doing?

I'd probably replace 4 with a slightly larger number (8 maybe) and put a comment like "the list halving shortcut is only useful for large lists" instead of the current comment, since it reads more easily. Alternatively, we could remove the if-statement all together to make the code even easier to read. How much performance impact does the duplicate work have? Is it noticeable?